WordPressKitTests/SiteCreationRequestEncodingTests.swift

```diff
@@ -0,0 +1,206 @@
+
+import XCTest
+@testable import WordPressKit
+
+class SiteCreationRequestEncodingTests: XCTestCase {
+
+    func testSiteCreationRequestEncoding_WithAllParameters_IsSuccessful() {
+        // Given
+        let request = SiteCreationRequest(
+            segmentIdentifier: 1,
+            verticalIdentifier: "p2v10",
+            title: "Come on in",
+            tagline: "This is a site I like",
+            siteURLString: "Cool Restaurant",
+            isPublic: true,
+            languageIdentifier: "TEST-ENGLISH",
+            shouldValidate: true,
+            clientIdentifier: "TEST-ID",
+            clientSecret: "TEST-SECRET"
+        )
+
+        // When
+        let encoder = JSONEncoder()
+
+        XCTAssertNoThrow(try encoder.encode(request))
+        let encodedJSON = try! encoder.encode(request)
+
+        XCTAssertNoThrow(try JSONSerialization.jsonObject(with: encodedJSON, options: []))
+        let serializedJSON = try! JSONSerialization.jsonObject(with: encodedJSON, options: [])
+
+        if let _ = serializedJSON as? [String : AnyObject] {} else {
+            XCTFail("Failed to encode a proper JSON dictionary!")
+        }
+        let jsonDictionary = serializedJSON as! [String : AnyObject]
+
+        // Then
+        XCTAssertNotNil(jsonDictionary["blog_name"])
+        XCTAssertNotNil(jsonDictionary["blog_title"])
+        XCTAssertNotNil(jsonDictionary["client_id"])
+        XCTAssertNotNil(jsonDictionary["client_secret"])
+        XCTAssertNotNil(jsonDictionary["public"])
+        XCTAssertNotNil(jsonDictionary["lang_id"])
+        XCTAssertNotNil(jsonDictionary["options"])
```

Comment:
Should we verify the values of the json dictionary, to ensure they contain what we expect them to? Also, should we verify the contents of the `options` dictionary here, as it has a couple of sublevels?